File: 2331-evaluate-boolean-binary-tree/2331-evaluate-boolean-binary-tree.py
class Solution:
    def evaluateTree(self, root: Optional[TreeNode]) -> bool:
        # since it is a full binary tree, then if any child is null, then it is a leaf node
        if root.left == None: return root.val == 1
        
        leftVal = self.evaluateTree(root.left)
        rightVal = self.evaluateTree(root.right)

        if root.val == 2: return (leftVal | rightVal)
        return (leftVal & rightVal)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

File: 2331-evaluate-boolean-binary-tree/2331-evaluate-boolean-binary-tree.go
func evaluateTree(root *TreeNode) bool {
    // since it is a full binary tree, then if any child is null, then it is a leaf node
    if root.Left == nil {
        return root.Val == 1
    }
    
    leftVal := evaluateTree(root.Left)
    rightVal := evaluateTree(root.Right)

    if root.Val == 2 {
        return (leftVal || rightVal)
    }
    return (leftVal && rightVal)
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */